### PR TITLE
CORTX-29869: Add rgw_s3 service for RGW component in cluster config file

### PR DESCRIPTION
**Pull request by tanujashinde0405**
## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds new functionality)
- [x] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [ ] Documentation additions or improvements
- [ ] Code quality improvements to existing code or test additions/updates

## Applicable issues

- This change fixes an issue: #
- This change is related to an issue: #

## CORTX image version requirements

This change requires the following images:

- `cortx-all:<version>`
- `cortx-rgw:<version>`

## Checklist

- [ ] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [x] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change addresses a CORTX Jira issue:

- [x] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)

### CODE REVIEW

**Review by tanujashinde0405 — COMMENTED** [review]
@keithpine , we need to merge : https://github.com/Seagate/cortx-hare/pull/2037 and this current PR together. Can we approve this PR and setup one call to merge this PR's together?

**Comment by osowski** [comment]
> @keithpine , we need to merge : [Seagate/cortx-hare#2037](https://github.com/Seagate/cortx-hare/pull/2037) and this current PR together. Can we approve this PR and setup one call to merge this PR's together?

We need to be moving away from developing such that "these PRs [in different repos] need to be merged together". If we need to have a larger discussion about what that looks like in the codebase to support more adaptive coding and interaction between components, I am happy to facilitate.

**Review by tanujashinde0405 — COMMENTED** [review]
@osowski @keithpine, We have merged the dependent hare PR: https://github.com/Seagate/cortx-hare/pull/2037, Are we good to merge this PR now?

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/_cluster.tpl [hunk 1]
```diff
@@ -33,6 +33,8 @@ cluster:
     - name: utils
     - name: hare
     - name: rgw
+      services:
+        - rgw_s3
```
Comment (keithpine) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/_cluster.tpl:37]: Any reason why the sample file has not been updated?

https://github.com/Seagate/cortx-prvsnr/blob/2fc7486a652db805619525fab5c26bed98d5a22b/conf/solution/cluster.yaml.sample#L36-L40

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/_config.tpl [hunk 2]
```diff
@@ -90,7 +90,7 @@ cortx:
       endpoints: {{- toYaml .Values.cortxMotr.confdEndpoints | nindent 6 }}
     clients:
     {{- if .Values.cortxRgw.enabled }}
-    - name: rgw
+    - name: rgw_s3
```
Comment (keithpine) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/_config.tpl:93]: Why is the significance of this name? It is not documented anywhere that I can find. The sample file has also not been updated for this.

https://github.com/Seagate/cortx-prvsnr/blob/2fc7486a652db805619525fab5c26bed98d5a22b/conf/solution/config.yaml.sample#L109

**Review by keithpine — APPROVED** [review]
This change works fine for me using image 725. There is still an issue of the provisioner sample files not reflecting the latest supported formats.

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/_cluster.tpl (hunk 1 above)
Comment (tanujashinde0405) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/_cluster.tpl:37]: @keithpine , We have made the sample files changes now for rgw_s3. Please check once.

Comment (keithpine) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/_cluster.tpl:37]: Looks good, thanks!

**Pull request merged**